framework/src/main/java/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -294,10 +294,8 @@ public S initialStore(UnderlyingAST underlyingAST, List<LocalVariableNode> param
         // Allow that local variables are retained; they are effectively final,
         // otherwise Java wouldn't allow access from within the lambda.
         // TODO: what about the other information? Can code further down be simplified?
-        // store.localVariableValues.clear();
         store.classValues.clear();
         store.arrayValues.clear();
-        store.methodValues.clear();
```

Comment:
I'm realizing now that a more precise way would be to only remove information about `methodValues` for which there is a side-effecting subexpression